configurator/configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/ResultImpl.java

```diff
@@ -71,12 +75,12 @@ public Status getStatus() {
     }
 
     @Override
-    public Optional<Throwable> getBadOutcome() {
-        return Optional.ofNullable(badOutcome);
+    public Optional<ConfiguratorException> getError() {
+        return Optional.ofNullable(error);
```

Comment:
I'm not sure how granular the exceptions will be. If there are nested exceptions, they'll certainly be there but walking the chain and determining root cause in a structured way may be difficult.